datafusion/physical-expr/src/utils.rs

```diff
@@ -911,12 +1064,18 @@ mod tests {
         ];
         let finer = Some(&finer[..]);
         let empty_schema = &Arc::new(Schema::empty());
```

Comment:
I wonder if some more unit tests would be valuable here mostly as a way to document how the functions were supposed to work -- trying to write more focused tests might also help be a forcing function to create fewer new functions / refactor functions to share more functionality

Reply:
I have added unit tests, for each normalization function.